cpp/src/arrow/flight/sql/odbc/tests/errors_test.cc

```diff
@@ -479,22 +481,35 @@ TYPED_TEST(ErrorsOdbcV2Test, TestSQLErrorEnvErrorFromDriverManager) {
   ASSERT_EQ(SQL_SUCCESS, SQLError(this->env, nullptr, nullptr, sql_state, &native_error,
                                   message, SQL_MAX_MESSAGE_LENGTH, &message_length));
 
-  EXPECT_GT(message_length, 50);
+  EXPECT_GT(message_length, 40);
 
   EXPECT_EQ(0, native_error);
 
   // Function sequence error state from driver manager
-  EXPECT_EQ(std::wstring(L"S1010"), std::wstring(sql_state));
+#ifdef _WIN32
+  // Windows Driver Manager returns S1010
+  EXPECT_EQ(SqlWcharToString(sql_state), kErrorStateS1010);
+#else
+  // unix Driver Manager returns HY010
+  EXPECT_EQ(SqlWcharToString(sql_state), kErrorStateHY010);
+#endif  // _WIN32
 
   EXPECT_FALSE(std::wstring(message).empty());
 }
 
+// TODO: verify that `SQLGetConnectOption` is not required by Excel.
+#ifndef __APPLE__
 TYPED_TEST(ErrorsOdbcV2Test, TestSQLErrorConnError) {
```

Comment:
Still need to verify if `SQLGetConnectOption` is required by latest version of Excel. Will remove this todo comment in future PRs.

For this PR, we are checking in the SQLErrors changes.